sentinelhub/aws_safe.py

```diff
@@ -33,34 +33,66 @@ def get_safe_struct(self):
         :return: nested dictionaries representing .SAFE structure
         :rtype: dict
         """
-        safe = {}
         main_folder = self.get_main_folder()
-        safe[main_folder] = {}
+        safe = {
+            main_folder: {
+                AwsConstants.AUX_DATA: {},
+                AwsConstants.DATASTRIP: self._get_datastrip_substruct(),
+                AwsConstants.GRANULE: self._get_granule_substruct(),
+                self.get_product_metadata_name(): self.get_url(AwsConstants.METADATA),
+                "INSPIRE.xml": self.get_url(AwsConstants.INSPIRE),
+                self.add_file_extension(AwsConstants.MANIFEST): self.get_url(AwsConstants.MANIFEST),
+                AwsConstants.HTML: {},  # AWS doesn't have this data
+                AwsConstants.INFO: {},  # AWS doesn't have this data
+            }
+        }
 
-        safe[main_folder][AwsConstants.AUX_DATA] = {}
+        if self.is_early_compact_l2a():
+            safe[main_folder]["L2A_Manifest.xml"] = self.get_url(AwsConstants.L2A_MANIFEST)
+            safe[main_folder][self.get_report_name()] = self.get_url(AwsConstants.REPORT)
 
-        safe[main_folder][AwsConstants.DATASTRIP] = {}
+        if self.safe_type == EsaSafeType.OLD_TYPE and self.baseline != "02.02":
+            safe[main_folder][_edit_name(self.product_id, "BWI") + ".png"] = self.get_url(
+                AwsConstants.PREVIEW, MimeType.PNG
+            )
+        return safe
+
+    def _get_datastrip_substruct(self):
+        """Builds a datastrip subfolder structure of .SAFE format."""
+        datastrip_safe = {}
         datastrip_list = self.get_datastrip_list()
         for datastrip_folder, datastrip_url in datastrip_list:
-            safe[main_folder][AwsConstants.DATASTRIP][datastrip_folder] = {}
-            safe[main_folder][AwsConstants.DATASTRIP][datastrip_folder][AwsConstants.QI_DATA] = {}
-            # S-2 L1C reports are on AWS only stored with tiles and without RADIOMETRIC_QUALITY
-            if self.has_reports() and self.data_collection is DataCollection.SENTINEL2_L2A:
+            datastrip_safe[datastrip_folder] = {}
+            datastrip_safe[datastrip_folder][AwsConstants.QI_DATA] = {}
```

Comment:
why not `datastrip_safe[datastrip_folder] = {AwsConstants.QI_DATA: {}}`?

Reply:
fixed